exasol_integration_test_docker_environment/lib/test_environment/spawn_test_container.py

```diff
@@ -38,11 +39,26 @@ def register_required(self):
         self.test_container_image_future = self.register_dependency(DockerTestContainerBuild())
         self.export_directory_future = self.register_dependency(CreateExportDirectory())
 
+    def is_reuse_possible(self) -> bool:
+        test_container_image_info = \
+            self.get_values_from_futures(self.test_container_image_future)["test-container"]  # type: ImageInfo
+        test_container = None
+        try:
+            test_container = \
+                self._client.containers.get(self.test_container_name)
+        except Exception as e:
+            pass
+        return self.network_info.reused and self.reuse_test_container and \
+               test_container is not None and \
+               test_container.image == test_container_image_info.get_target_complete_name() and \
+               test_container_image_info.image_state == ImageState.USED_LOCAL
```

Comment:
Do you really need to check that? Wouldn't docker build it automatically if not locally available?

Reply:
besides build and used_local there are other options, for example load or pull, and I only want to reuse when it was locally available